pkg/controllers/cstorvolumeconfig/policy.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cstorvolumeconfig
+
+import (
+	apis "github.com/openebs/api/pkg/apis/cstor/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+var (
+	// defaultQueueDepth represents the queue size at iSCSI target which limits
+	// the ongoing IO count from client.
+	defaultQueueDepth = "32"
+	// defaultIOWorkers represents default (luWorker) number of threads that are
+	// working on queue
+	defaultIOWorkers = int64(6)
+)
+
+type policyOptFuncs func(*apis.CStorVolumePolicySpec, apis.CStorVolumePolicySpec)
+
+// validatePolicySpec validates the provided policy created by the user and
+// otherwise sets the defaults policy spec for cstor volumes.
+func validatePolicySpec(policy *apis.CStorVolumePolicySpec) {
```

Comment:
`validatePolicySpec` means it should validate right? but it is doing defaulting...

Reply:
Will be adding more validation based on the invalid value later , hence the name